src/main/java/christmas/domain/Order.java

```diff
@@ -0,0 +1,58 @@
+package christmas.domain;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Order {
+
+    private static final int MAX_QUANTITY = 20;
+    private static final String INSUFFICIENT_ORDERLINE_ERROR_MESSAGE = "[ERROR] 주문 항목이 비어 있습니다.";
+    private static final String TOO_MANY_ORDERLINE_QUANTITY_ERROR_MESSAGE = "[ERROR] 메뉴는 한 번에 최대 20개까지만 주문할 수 있습니다.";
+    private static final String ONLY_DRINKS_ERROR_MESSAGE = "[ERROR] 음료만 주문할 수 없습니다.";
+
+    private List<OrderLine> orderLines;
+
+    public Order(List<OrderLine> orderLines) {
+        validateAtLeastOneOrderLine(orderLines);
+        validateTotalQuantity(orderLines);
+        validateOnlyDrinks(orderLines);
+        this.orderLines = orderLines;
+    }
+
+    public Money getTotalAmounts() {
+        return orderLines.stream()
+            .map(OrderLine::getAmounts)
+            .reduce(Money::plus)
+            .get();
+    }
+
+    private void validateAtLeastOneOrderLine(List<OrderLine> orderLines) {
+        if (orderLines == null || orderLines.isEmpty()) {
+            throw new IllegalArgumentException(INSUFFICIENT_ORDERLINE_ERROR_MESSAGE);
+        }
+    }
+
+    private void validateTotalQuantity(List<OrderLine> orderLines) {
+        int totalQuantity = orderLines.stream()
+            .mapToInt(OrderLine::getQuantity)
+            .sum();
+
+        if (totalQuantity > MAX_QUANTITY) {
+            throw new IllegalArgumentException(TOO_MANY_ORDERLINE_QUANTITY_ERROR_MESSAGE);
+        }
+    }
+
+    private void validateOnlyDrinks(List<OrderLine> orderLines) {
+        if (orderLines.stream().allMatch(OrderLine::isDrink)) {
```

Comment:
```suggestion
        if (isOnlyDrink())
```
메서드로 분리해서 가독성을 높이는건 어떨까요?

Reply:
이미 validateOnlyDrinks 라는 메서드 내에서 또 다른 private 메서드로 또 빼게 되면 의미의 중복이 생길 것 같아 굳이 빼지는 않았습니다. 말씀해주신 부분도 다시 고려해보겠습니다 :>

Reply:
메서드 분리가 부담스러우시다면 로컬 변수로 분리해줘도 좋겠군요 ㅋㅋ

```
    private void validateOnlyDrinks(List<OrderLine> orderLines) {
        boolean containDrink = orderLines.stream().allMatch(OrderLine::isDrink);
        
        if (containDrink) {
            throw new IllegalArgumentException(ONLY_DRINKS_ERROR_MESSAGE);
        }
    }

```